src/AggregateFunctions/AggregateFunctionSum.h

```diff
@@ -20,11 +20,72 @@ struct AggregateFunctionSumData
 {
     T sum{};
 
-    void add(T value)
+    void ALWAYS_INLINE add(T value)
     {
         sum += value;
     }
 
+    /// Vectorized version
+    template <typename Value>
+    void NO_INLINE addMany(const Value * __restrict ptr, size_t count)
+    {
+        /// Compiler cannot unroll this loop, do it manually.
+        /// (at least for floats, most likely due to the lack of -fassociative-math)
```

Comment:
Why not add `-fassociative-math` then?

Reply:
It will make Kahan summation algorithm to have no effect.

But we can enable
```
-fno-math-errno
-fno-rounding-math
-fno-signaling-nans
-fno-signed-zeros
-fno-trapping-math
-fassociative-math
-freciprocal-math
```
for all codebase and disable it whenever something like Kahan summation is used.
More care should be taken to figure out these places.